From david-b@pacbell.net Mon Nov  7 15:32:37 2005
Message-ID: <20051107233125.97726.qmail@web80709.mail.yahoo.com>
Date: Mon, 7 Nov 2005 15:31:25 -0800 (PST)
From: David Brownell <david-b@pacbell.net>
Subject: USB: wakeup flag updates (1/3) sl811-hcd
To: greg@kroah.com


This makes the SL811 HCD use the driver model wakeup flags for its
controller, not the flags in the HCD glue (which will be removed).

From: David Brownell <dbrownell@users.sourceforge.net>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

 drivers/usb/host/sl811-hcd.c |    6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

--- gregkh-2.6.orig/drivers/usb/host/sl811-hcd.c	2005-11-11 22:48:28.000000000 -0800
+++ gregkh-2.6/drivers/usb/host/sl811-hcd.c	2005-11-12 14:14:27.000000000 -0800
@@ -1581,7 +1581,9 @@
 	hcd->state = HC_STATE_RUNNING;
 
 	if (sl811->board) {
-		hcd->can_wakeup = sl811->board->can_wakeup;
+		if (!device_can_wakeup(hcd->self.controller))
+			device_init_wakeup(hcd->self.controller,
+				sl811->board->can_wakeup);
 		hcd->power_budget = sl811->board->power * 2;
 	}
 
@@ -1805,7 +1807,7 @@
 	 * let's assume it'd only be powered to enable remote wakeup.
 	 */
 	if (dev->dev.power.power_state.event == PM_EVENT_SUSPEND
-			|| !hcd->can_wakeup) {
+			|| !device_can_wakeup(&hcd->self.root_hub->dev)) {
 		sl811->port1 = 0;
 		port_power(sl811, 1);
 		return 0;
